### PR TITLE
Feature/213 solidify badge statuses 5

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Fix issue https://github.com/unfoldingWord-dev/door43.org/issues/213 solidify badge statuses 5

Changes:
Fix to return errors and warnings generated by OBSInspection of each file.
Md2HtmlConverter - added check for missing chapters. 
TestMd2HtmlConverter - added test cases for empty chapter, missing chunk, missing chapter.

### CODE REVIEW

File: tests/converter_tests/test_md2html_converter.py [hunk 1]
```diff
@@ -81,5 +86,156 @@ def test_run(self):
...
+        self.expected_info_empty = False;
+        missing_chapters = [1]
+
+        # when
+        tx = self.doTransformObs(file_name)
+
+        #then
+        self.verifyTransform(tx, missing_chapters)
+
+
```
Comment (richmahn) [inline tests/converter_tests/test_md2html_converter.py]: Let's do one blank line between functions in a class

File: tests/converter_tests/test_md2html_converter.py [hunk 2]
```diff
@@ -81,5 +86,156 @@ def test_run(self):
...
+        self.expected_info_empty = False;
+        missing_chapters = []
+
+        # when
+        tx = self.doTransformObs(file_name)
+
+        #then
+        self.verifyTransform(tx, missing_chapters)
+
+
```
Comment (richmahn) [inline tests/converter_tests/test_md2html_converter.py]: Ditto here about one blank line

File: converters/md2html_converter.py [hunk 3]
```diff
@@ -22,14 +23,17 @@ def convert_obs(self):
...
         for filename in files:
             if filename.endswith('.md'):
-                # Convert files tat are markdown files
+                # Convert files that are markdown files
                 with codecs.open(filename, 'r', 'utf-8-sig') as md_file:
                     md = md_file.read()
                 html = markdown.markdown(md)
                 html = html_template.safe_substitute(content=html)
                 base_name = os.path.splitext(os.path.basename(filename))[0]
+                found_chapters[base_name] = True
```
Comment (richmahn) [inline converters/md2html_converter.py:36]: Where is this used later?

**Review by richmahn — CHANGES_REQUESTED** [review]
See my line comments

File: converters/md2html_converter.py (hunk 3 above)
Comment (PhotoNomad0) [reply converters/md2html_converter.py:36]: @richmahn - Below, starting in line 68.

File: tests/converter_tests/test_md2html_converter.py (hunk 1 above)
Comment (PhotoNomad0) [reply tests/converter_tests/test_md2html_converter.py]: sure

**Review by richmahn — APPROVED** [review]

File: tests/door43_tools_tests/test_usfm_preprocessor.py [hunk 4]
```diff
@@ -48,11 +48,35 @@ def test_TsUsfmMarkdownPreprocessorComplete(self):
...
+            meta = MetaData(meta_path)
+        manifest = Manifest(file_name=manifest_path, repo_name=repo_name, files_path=repo_dir, meta=meta)
+        return manifest, repo_dir, temp_dir
+
     def runTsUsfmMarkdownPreprocessor(self, manifest, repo_dir):
-        self.out_dir = tempfile.mkdtemp(prefix='output_')
-        compiler = TsBibleUsfmPreprocessor(manifest, repo_dir, self.out_dir)
-        compiler.run()
-        return self.out_dir
+            self.out_dir = tempfile.mkdtemp(prefix='output_')
```
Comment (phillip-hopper) [inline tests/door43_tools_tests/test_usfm_preprocessor.py]: The indentation looks wrong here.

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

File: tests/door43_tools_tests/test_usfm_preprocessor.py (hunk 4 above)
Comment (PhotoNomad0) [reply tests/door43_tools_tests/test_usfm_preprocessor.py]: oops

**Comment by PhotoNomad0** [comment]
@phillip-hopper - fixed what you found

**Review by phillip-hopper — APPROVED** [review]
LGTM 👍

**Pull request merged**